CONTRIBUTING.md

```diff
@@ -1,42 +1,229 @@
-first, please read through the rest of this doc.
+We welcome contributions to OpenHTF! Provided you've satisfied our legal
+requirements, contributing is as easy as submitting a pull request with your
+changes in order to kick off the review process. But first, please read through
+the rest of this doc to help ensure a smooth code review.
 
 
 ## Legal Requirements
 In order to become a contributor, you first need to sign the appropriate
-[Contributor License Agreement](https://cla.developers.google.com/clas).
+[Contributor License Agreement](https://cla.developers.google.com/clas) (if you happen to be a Google employee, you're already covered).
 
 
-## Managing Dependencies
+## Code Reviews
+All contributions to OpenHTF, including those made by official maintainers, must
+go through code review. Code reviews ensure that the codebase stays as easily
+maintainable as possible, and that contributions fully align with the stated
+goals of OpenHTF.
+
+
+### Process Overview
+Our code review process generally follows a predictable flow:
+1. Fork the OpenHTF repository on github.
+2. Make commits to your fork.
+3. Submit a pull request (PR) to pull your fork into the official repo.
+4. OpenHTF maintainers will review the PR and make comments.
+5. Discuss and make more commits to your fork to address reviewer feedback.
+6. Update the PR discussion on github to indicate readiness for another look
+   from the reviewers.
+7. Repeat steps 4-6 until your PR receives an **LGTM** ("Looks Good To Me") from 
+   at least two official maintainers.
+8. One of the maintainers will merge the PR into the main repo.
+
+
+### Code Standards
+For a smooth code review, it helps to make sure your code adheres to standards,
+conventions, and design goals for OpenHTF. A best-effort attempt to understand 
+and meet these standards before requesting code review can go a long way towards
+making the review process as fast and painless as possible.
+
+
+#### Follow Style Guidelines
+OpenHTF's Python code follows the
+[Google Python Style Guide](https://google.github.io/styleguide/pyguide.html).
+We provide a `pylintrc` file at the top level of our repo that you can use with
+[pylint](https://www.pylint.org/) to lint your Python code.
+
+OpenHTF's built-in web frontend, written in TypeScript and using Angular2,
+follows the [Angular2 Style Guide](https://angular.io/styleguide).
+We recommend using [codelyzer](https://www.npmjs.com/package/codelyzer) to lint
+your frontend code.
+
+
+#### Provide Documentation
+OpenHTF code is fairly well-documented through the liberal use of docstrings and
+comments in the actual code. If your code is doing something that isn't
+immediately intuitive, consider adding clarifying comments.
+
+In addition to inline documentation, adding example scripts to the `examples/`
+directory is a great way to express the intended usage of OpenHTF features.
+
+
+#### Limit Complexity
```

Comment:
I would include a sentence in this section indicating that minimizing code complexity is a core tenet of the design philosophy of OpenHTF.  I would consider it even higher priority than performance in most cases, and the reason for this is primarily that reliability and stability of hardware tests are often (far) more important than a few seconds of execution time.  This is a fairly application-specific attribute of this software over, for example, more computationally intensive or latency-sensitive applications.